elastic/kibana
# Fix incorrect no data alert for Metric threshold and Custom threshold rules when groupBy field is added and removed

Fixes #168829

## Summary

This PR fixes the issue when a Metric threshold or Custom threshold rule is created with the groupBy field and then gets updated, and the groupBy field is removed. In this scenario, if no alert is triggered, a no-data alert is generated wrongly.
This issue is due to saving `groupBy` field as an empty string in the rule's saved object.

<img width="326" alt="Screenshot 2023-10-16 at 15 28 41" src="https://github.com/elastic/kibana/assets/12370520/458e1a94-7234-4fa5-89bd-afb8a8b5f158">

### 🧪 How to test
- Create a Metric threshold or Custom threshold rule with `groupBy` field that does not trigger any alert
- Update the rule and remove the groupBy field
- You shouldn't see a no-data alert in this case anymore
